pandas/tests/series/indexing/test_where.py

```diff
@@ -38,23 +38,24 @@ def test_where_unsafe_float(float_numpy_dtype):
 
 
 @pytest.mark.parametrize(
-    "dtype,expected_dtype",
+    "dtype,expected_dtype,warn",
     [
-        (np.int8, np.float64),
-        (np.int16, np.float64),
-        (np.int32, np.float64),
-        (np.int64, np.float64),
-        (np.float32, np.float32),
-        (np.float64, np.float64),
+        (np.int8, np.float64, FutureWarning),
+        (np.int16, np.float64, FutureWarning),
+        (np.int32, np.float64, FutureWarning),
+        (np.int64, np.float64, FutureWarning),
+        (np.float32, np.float32, None),
+        (np.float64, np.float64, None),
```

Comment:
can the `warn` variable be set inside the function?  i think that can make the reasoning for how it is defined clearer (e.g `warn = None if expected_dtype.kind == dtype.kind == "f" else FutureWarning`)